docs/proposals/gateway-settings.md

```diff
@@ -0,0 +1,171 @@
+
+## Non-Goals
+
+- Provide implementation details for implementing the Gateway settings.
+
+## Introduction
+
+### Gateway Settings
+
+Gateway Settings are NGINX directives or configuration attached at the GatewayClass level that should be solely controlled by the Cluster Operator and should not be changed by the Application Developers. All Gateways attached to this GatewayClass will inherit these settings.
+
+These settings apply to the `main`, `http`, and/or `stream` contexts of the NGINX config.
+
+To begin, the Gateway Settings config will include the following NGINX directives (focusing on OpenTelemetry tracing):
+
+- [`otel_exporter`](https://nginx.org/en/docs/ngx_otel_module.html#otel_exporter)
+- [`otel_service_name`](https://nginx.org/en/docs/ngx_otel_module.html#otel_service_name)
+- [`otel_span_attr`](https://nginx.org/en/docs/ngx_otel_module.html#otel_span_attr): set global span attributes that will be merged with the span attributes set in the [Observability extension](nginx-extensions.md#observability).
+
+In the future, this config will be extended to support other directives, such as those defined in the [NGINX Extensions Proposal](nginx-extensions.md#gateway-settings).
+
+## API, Customer Driven Interfaces, and User Experience
+
+The `GatewaySettings` API is a CRD that is a part of the `gateway.nginx.org` Group. It will be referenced in the `parametersRef` field of a GatewayClass. It will live at the cluster scope.
+
+This is a dynamic configuration that can be changed by a user at any time, and NGF will propagate those changes to NGINX. This is something we need to clearly document in our public documentation about this feature, so that users know that all Gateways under the Class can be updated by these settings.
+
+For example, a `GatewaySettings` named `gw-settings` would be referenced as follows:
+
+```yaml
+kind: GatewayClass
+metadata:
+    name: nginx
+spec:
+    controllerName: gateway.nginx.org/nginx-gateway-controller
+    parametersRef:
+        group: gateway.nginx.org/v1alpha1
+        kind: GatewaySettings
+        name: gw-settings
+```
+
+Below is the Golang API for the `GatewaySettings` API:
+
+### Go
+
+```go
+package v1alpha1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+type GatewaySettings struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec defines the desired state of the GatewaySettings.
+	Spec GatewaySettingsSpec `json:"spec"`
+}
+
+type GatewaySettingsSpec struct {
+	// Telemetry specifies the OpenTelemetry configuration.
+	// +optional
+	Telemetry *Telemetry `json:"telemetry,omitempty"`
+}
+
+// Telemetry specifies the OpenTelemetry configuration.
+type Telemetry struct {
+	// Exporter specifies OpenTelemetry export parameters.
+	// +optional
+	Exporter *Exporter `json:"exporter,omitempty"`
+
+	// ServiceName is the "service.name" attribute of the OpenTelemetry resource.
+	// Default is 'nginx-gateway-fabric:<gateway-name>'.
```

Comment:
Thinking ahead for when we support multiple gateways, should this be gateway-class name?

Reply:
The Gateway will ultimately be the one generating the span, so I figured that that it should reference the origin.

Reply:
@kate-osborn good point
@pleshakov Yeah, we should also probably do that

The name is getting a bit long, so I could do `Default: ngf:namespace:gateway`, otherwise this is the prefix.

Reply:
sounds good to me